src/adr/0051-email-rate-limiting.md

```diff
@@ -0,0 +1,40 @@
+# Email rate limiting
+
+Date: 2021-01-10
+
+## Status
+
+// todo
+
+## Context
+
+Rate limiting is one of the most popular features of any service. They limit how many emails you can send/receive from/to
+each email account over a given period of time.  
+We believe the rate-limiting will help James has more control of the resource and easy to dynamic config the user policy.
+
+## Decision
+
+Provide 2 criteria to limit to each duration:
+
+- Number of emails
+- Total size of emails
+
+This can be done via mailets (`PerSenderRateLimit` and `PerRecipientRateLimit`). Depending on the position in the pipeline this could allow rating limit all emails in
+transit, relayed emails or locally delivered emails.    
+The rate limit will be config in [mailetcontainer.xml](/server/apps/distributed-app/sample-configuration/mailetcontainer.xml).
+
+Provide the interface `RateLimiter`, that will evaluate the current rate limit and return result acceptable or exceeded.
+The default implement of RateLimiter will be using [Redis](https://redis.io) to store the current rate of each email (Redis is an in-memory data structure store, used
+as a distributed).
```

Comment:
Those mailets will be based on a generic `RateLimiter` interface. We will propose two implementations for it:
 - In memory (guava based) suitable for single instance deployments
 - Redis based, suitable for distributed deployments.
 
 The implementation chosen will be configurable as part of mailet configuration.